### PR TITLE
Upstream sync 60/N: merge 76fedaa2a5 (36 commits, conflict-free)

### PR DESCRIPTION
**Pull request by roberteg16**
## Summary

Conflict-free upstream batch: the 36 commits between `95ed0feaa5` and `76fedaa2a5` (2026-07-10 04:34 UTC .. 2026-07-11 13:56 UTC). `git merge` reported no conflicts; nothing here is a manual resolution.

142 files, +5675/−1921. `csrc/rocm/` untouched, so no `SKINNY=1`. One file deleted — `tests/models/multimodal/generation/test_vit_backend_functionality.py`, together with the helpers only it used.

## The one that needed a closer look

`DeepseekV4ForCausalLM` and `DeepseekV4Model` appeared in the removed-definition list, and the ROCm implementation of that model is the fork-relevant one, so this was worth reading rather than counting.

They are not removed — they **gain mixins**:

```
class DeepseekV4Model(nn.Module)                   ->  (nn.Module, EagleModelMixin)
class DeepseekV4ForCausalLM(nn.Module, SupportsPP) ->  (..., SupportsEagle3)
```

Confirmed both mixins are imported in the same file (`amd/model.py:44-45`). That check matters more than usual: **a missing name in a base-class list raises at import time, and `py_compile` only parses** — it would not have caught it, and neither would a smoke test that never imports that model.

The platform dispatch in `vllm/models/deepseek_v4/__init__.py` still routes ROCm to `.amd.model`, and `registry.py:94` still points at the package.

`combine_topk_swa_indices_ragged`, `_remap_mistral_to_ds` and `_reorder_gate_up_to_up_gate` have zero references tree-wide.

Boundary probed in a ladder: k=608 clean, k=609 conflicts, and k=610 through k=1876 all conflict.

The next commit, `51878e5b6e` "[2/N][KV-Cache Layout Refactor] Pack K/V into the content dim across attention backends (#44455)", conflicts and gets its own PR — a KV-cache layout refactor across attention backends, so that one will need care.

**Stacked on** #1167. **Merge commit only — do not squash or rebase.**

AI assistance was used to prepare this merge.

## Test plan

- [x] `git merge` clean, no manual resolution
- [x] DeepseekV4 mixin additions verified, including that both mixin names are imported
- [x] ROCm platform dispatch and registry entry confirmed intact
- [x] `py_compile` over all 64 changed Python files
- [x] pre-commit (ruff, mypy 3.10, SPDX, forbidden-imports)
- [x] Correctness — covered by CI (`test-kernels-correctness`)
- [x] Build + 5-benchmark sweep vs the dashboard incl. the AWQ canary

### CODE REVIEW

**Comment by roberteg16** [comment]
## Verification (re-measured)

Build clean. Correctness runs in CI (`test-kernels-correctness`); this is the five model benchmarks. All five sanity checks passed.

**This is a re-run.** The first measurement of this batch was taken while an unniced `pre-commit --all-files` was running on the same box, which inflated `bench1` (79.3 tok/s, TTFT 191.8 ms). These are the clean numbers.

| Test | TTFT (ms) | dashboard | Δ | Decode (tok/s) | dashboard | Δ |
|---|---|---|---|---|---|---|
| Qwen3-30B-A3B-Instruct-2507-AWQ-4bit_128 | 189.2 | 189.6 | −0.2% | 79.8 | 80.1 | −0.4% |
| Qwen3.6-35B-A3B-W4A16 VLM +int8 lm-head | 265.0 | 283.0 | −6.4% | 84.0 | 83.7 | +0.4% |
| Gemma-3-4B-IT_VLM_w4a16 | 472.8 | 465.0 | +1.7% | 63.7 | 63.6 | +0.2% |
| Qwen3-Omni-30B-A3B-Instruct_VLM_AWQ-4bit | 682.2 | 689.8 | −1.1% | 75.7 | 75.7 | ±0.0% |
| Qwen2.5-0.5B-Instruct-AWQ_128 (AWQ canary) | 17.4 | — | — | 355.8 | 354.8 | +0.3% |

**No regression** — decode within ±0.4% on all five.

### This run is also the "A" half of the #1169 regression A/B

Batch 60 is the **last good** batch before the Gemma-3 decode drop tracked on #1169. Rebuilt and re-measured today, hours after the originals and in the same environment, `bench3` comes back at **63.7 tok/s / 15.70 ms TPOT** — matching b58/b59/b60 exactly.

That rules out environmental drift or machine state as the explanation for the b61/b62/b63 readings of 61.7 / 61.7 / 61.5 (TPOT 16.2). The difference is in the code. `b61redo` is building now as the "B" half.

The canary's TTFT baseline is stale (predates `c4fd9794e9`, which removed `AsyncMicrobatchTokenizer`), so no delta is shown — ~16–17 ms is the current normal.

**Comment by eble-amd** [comment]
Retrospective review:

```bash
git diff a32e88ff349a70bc06944aaa49b09dfa222a6605{^,} > pr-diff.txt
git diff 95ed0feaa5 76fedaa2a5 > upstream-diff.txt
meld upstream-diff.txt pr-diff.txt
```

looks as advertised
